test/indexer/test_tdr.py

```diff
@@ -271,13 +285,54 @@ def test_list_bundles(self):
                                          for links_id in links_ids
                                      ])
-        bundle_ids = self.plugin.list_bundles(source, prefix='42')
-        bundle_ids.sort(key=attrgetter('uuid'))
-        self.assertEqual(bundle_ids, [
+        expected_bundle_ids = [
             TDRBundleFQID(source=source, uuid='42-abc', version=current_version),
             TDRBundleFQID(source=source, uuid='42-def', version=current_version),
             TDRBundleFQID(source=source, uuid='42-ghi', version=current_version)
-        ])
+        ]
+        with self.subTest('list_bundles'):
+            actual_bundle_ids = self.plugin.list_bundles(source, prefix='42')
+            actual_bundle_ids.sort(key=attrgetter('uuid'))
+            self.assertEqual(expected_bundle_ids, actual_bundle_ids)
+        with self.subTest('count_bundles_unpartitioned'):
+            unpartitioned_src = attr.evolve(source, prefix=None)
+            actual_bundle_count = self.plugin.count_bundles(unpartitioned_src)
+            self.assertEqual(len(links_ids), actual_bundle_count)
+        with self.subTest('count_bundles_partitioned'):
+            partitioned_src = attr.evolve(source, prefix=Prefix.parse('42/0'))
+            actual_bundle_count = self.plugin.count_bundles(partitioned_src)
+            self.assertEqual(len(expected_bundle_ids), actual_bundle_count)
+
+    def test_list_and_count_files(self):
+        source = self.source
+        self._make_mock_tdr_tables(source)
+        prefix = '3'
+        tables = self._load_canned_file_version(uuid=source.id,
+                                                version=None,
+                                                extension='tables.tdr')['tables']
+        all_files = [
+            row['descriptor']
+            for table_name, table_contents in tables.items()
+            if table_name.endswith('_file')
+            for row in table_contents['rows']
+        ]
+        partition_file_uuids = [
+            file['file_id']
+            for file in all_files
+            if file['sha256'].startswith(prefix)
+        ]
+        with mock.patch.object(tdr_hca.Plugin, '_assert_partition'):
+            with self.subTest('list_files'):
+                actual_file_uuids = [file.uuid for file in self.plugin.list_files(source, prefix=prefix)]
+                self.assertEqual(partition_file_uuids, actual_file_uuids)
+        with self.subTest('count_files_unpartitioned'):
+            unpartitioned_src = attr.evolve(source, prefix=None)
+            actual_file_count = self.plugin.count_files(unpartitioned_src)
+            self.assertEqual(len(all_files), actual_file_count)
+        with self.subTest('count_files_partitioned'):
+            partitioned_src = attr.evolve(source, prefix=Prefix.parse(f'{prefix}/0'))
+            actual_file_count = self.plugin.count_files(partitioned_src)
+            self.assertEqual(len(partition_file_uuids), actual_file_count)
```

Comment:
```suggestion
            self.assertGreater(actual_file_count, 0)
            self.assertEqual(len(partition_file_uuids), actual_file_count)
```

If the canned source was ever changed, we might not know this subtest found zero files.